iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -432,6 +448,12 @@ public override async Task SendMethodResponseAsync(DirectMethodResponse methodRe
                     throw new IotHubClientException($"Failed to send direct method response with reason code {result.ReasonCode}", true);
                 }
             }
+            catch (MqttCommunicationTimedOutException e)
+            {
+                // MQTTNet throws MqttCommunicationTimedOutException instead of OperationCanceledException
+                // when the cancellation token requests cancellation.
+                throw new OperationCanceledException("Timed out waiting for MQTT message to be acknowledged.", e);
```

Comment:
Might make the string a constant, or even make throwing this method a private helper so you don't have to repeat it so much.